shell/utils/validators/formRules/index.ts

```diff
@@ -170,7 +170,18 @@ export default function(
 
   const url: Validator = (val: string) => val && !isUrl(val) ? t('validation.setting.serverUrl.url') : undefined;
 
-  const gitRepository: Validator = (val: string) => val && !/^((http|git|ssh|http(s)|file|\/?)|(git@[\w\.]+))(:(\/\/)?)([\w\.@\:\/\-]+)([\d\/\w.-]+?)(.git){0,1}(\/)?$/gm.test(val) ? t('validation.git.repository') : undefined;
+  const gitRepository: Validator = (url: string) => {
```

Comment:
This could be simplified by just removing the %20 chars from `url` up front. 

Positives: It's a lot cleaner and a smidge quicker
Negatives: encoded whitespace char in schema, host, etc would be allowed 

Another option would be to allow `%` in the capture group `([\w\.@\:\/\-]+)`, though that applies to the host as well as path and doesn't check for digits afterwards. Maybe the worst of the three.

We need to come back and visit this anyway (to only accept encoded chars path and later, and is a file a valid git url?)

Reply:
Yes, here I decided to test the two chunks of URL separately to avoid accepting the %20 in the protocol.
The regex is generic, the `file` would be for URL scheme that refers to a file in local machine, definitely not Fleet usage, it can be removed.
I would keep it like this and improve it later.

Reply:
My reasoning was this solution is more complex and possibly unsafe, which isn't good for a temporary quick fix.

It looks like

`!urlPart2` - is it JUST part1 (part2 missing)
`url === urlPart2` - is it JUST part2 (part1 missing)
`regexPart2.test` - is part2 valid

I've given it a test and bar the fix it does look like it works. 

Can you create an issue to reduce complexity and fix the issue of the white space in the host part of the url?

FWIW these invalid scenarios fail in both this PR and our current code. If we're looking at fixing this properly we should also cover them
`https://https://`
`http:/ww.abc.com`, `http:ww.abc.com`
`git@github.com:a/`

Reply:
part1 (the scheme) is tested by `const urlPart2 = url.replaceAll(regexPart1, '');` + `url === urlPart2` . If both are verified it means part1 fails on scheme validation via regexPart1 regex.

Nevertheless, I agree in revisiting the validation to reduce the complexity and also because your test cases failed with version in master, so definitely to improve.

I'm meging current PR and creating the new issue.